atomics/include/desul/atomics/Generic.hpp

```diff
@@ -43,6 +43,8 @@ struct MinOper {
   }
 };
 
+// This exit early optimization causes weird compiler errors with MSVC 2019
+#ifndef DESUL_HAVE_MSVC_ATOMICS
```

Comment:
Why didn't you just hide the specialization of `may_exit_early` then so it is always false for MSVC?

Reply:
didn't think of that. But I can try.

Reply:
Ping